src/nft/components/sell/modal/ListingButton.tsx

```diff
@@ -0,0 +1,256 @@
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { ArrowRightIcon, HazardIcon, LoadingIcon, XMarkIcon } from 'nft/components/icons'
+import { bodySmall } from 'nft/css/common.css'
+import { useNFTList, useSellAsset } from 'nft/hooks'
+import { ListingStatus, WalletAsset } from 'nft/types'
+import { pluralize } from 'nft/utils/roundAndPluralize'
+import { useEffect, useMemo, useRef, useState } from 'react'
+
+import * as styles from './ListingModal.css'
+import { getListings } from './utils'
+
+interface ListingButtonProps {
+  onClick: () => void
+  buttonText: string
+  showWarningOverride?: boolean
+}
+
+export const ListingButton = ({ onClick, buttonText, showWarningOverride = false }: ListingButtonProps) => {
+  const sellAssets = useSellAsset((state) => state.sellAssets)
```

Comment:
I haven't use zustand a ton, but is there a specific reason we don't combine all these variables into 

const {sellAssets, addMarketplaceWarning, etc. } = useSellAssset()

Reply:
you can do it like this const { a, b } = useSellAssert((state) => ({a: state.a, b: state.b))

Reply:
Having the lines individually is the most efficient for zustand's equality check and previously was the only option. They have since added a method for doing multiple state picks using the shallow equality method, however the single line method is still the most efficient according to their docs and I would prefer to stick with it as a convention. More reading: https://github.com/pmndrs/zustand#selecting-multiple-state-slices